src/api/cli/integration/integration_test.go

```diff
@@ -63,11 +66,93 @@ func TestGenerateAddresses(t *testing.T) {
 	var w wallet.ReadableWallet
 	loadJSON(t, wltPath, &w)
 
+	golden := filepath.Join("testdata", "generateAddresses.golden")
+	if *update {
+		writeJSON(t, golden, w)
+	}
+
 	var expect wallet.ReadableWallet
-	loadJSON(t, filepath.Join("golden", "generateAddresses.golden"), &expect)
+	loadJSON(t, golden, &expect)
 	require.Equal(t, expect, w)
 }
 
+func TestVerifyAddress(t *testing.T) {
+	if !doLiveOrStable(t) {
+		return
+	}
+
+	output, err := exec.Command(binaryPath, "verifyAddress", "2Kg3eRXUhY6hrDZvNGB99DKahtrPDQ1W9vN").CombinedOutput()
+	require.NoError(t, err)
+	require.Empty(t, output)
+}
+
+func TestStableListAddress(t *testing.T) {
+	if !doStable(t) {
+		return
+	}
+
+	output, err := exec.Command(binaryPath, "listAddresses").CombinedOutput()
+	require.NoError(t, err)
+
+	var wltAddresses struct {
+		Addresses []string `json:"addresses"`
+	}
+	json.Unmarshal(output, &wltAddresses)
+
+	golden := filepath.Join("testdata", "listAddresses.golden")
+	if *update {
+		writeJSON(t, golden, wltAddresses)
+	}
+
+	var expect struct {
+		Addresses []string `json:"addresses"`
+	}
+	loadJSON(t, golden, &expect)
+	require.Equal(t, expect, wltAddresses)
+}
+
+func TestStableAddressBalance(t *testing.T) {
+	if !doStable(t) {
+		return
+	}
+
+	output, err := exec.Command(binaryPath, "addressBalance", "2kvLEyXwAYvHfJuFCkjnYNRTUfHPyWgVwKt").CombinedOutput()
+	require.NoError(t, err)
+
+	var addrBalance cli.BalanceResult
+	json.Unmarshal(output, &addrBalance)
+
+	golden := filepath.Join("testdata", "addressBalance.golden")
+	if *update {
+		writeJSON(t, golden, addrBalance)
+	}
+
+	var expect cli.BalanceResult
+	loadJSON(t, golden, &expect)
+	require.Equal(t, expect, addrBalance)
+}
+
+func TestStableWalletBalance(t *testing.T) {
+	if !doStable(t) {
+		return
+	}
+
+	output, err := exec.Command(binaryPath, "walletBalance").CombinedOutput()
+	require.NoError(t, err)
+
+	var wltBalance cli.BalanceResult
+	json.Unmarshal(output, &wltBalance)
+
+	golden := filepath.Join("testdata", "walletBalance.golden")
+	if *update {
+		writeJSON(t, golden, wltBalance)
+	}
+
+	var expect cli.BalanceResult
+	loadJSON(t, golden, &expect)
+	require.Equal(t, expect, wltBalance)
+}
+
```

Comment:
For live tests, will have to specify wallet file through an env var, and check that the exit status was fine and the output is parseable

Reply:
We already set a wallet in `TestMain`. I have gone ahead and used that only for doing live tests.

https://github.com/therealssj/skycoin/blob/9d1be7c5308546beb7eb024e2006333ba3f1575f/src/api/cli/integration/integration_test.go#L511